cpp/src/arrow/builder.h

```diff
@@ -87,7 +87,9 @@ class ARROW_EXPORT ArrayBuilder {
   /// been appended. Does not account for reallocations that may be due to
   /// variable size data, like binary values. To make space for incremental
   /// appends, use Reserve instead.
-  /// \param[in] capacity the minimum number of additional array values
+  ///
+  /// \param[in] capacity the minimum number of additional array values. Must
```

Comment:
I think the "additional" here is misleading; I think this needs to be "the number of total values to allocate memory for". I'll make this minor change